parser/src/parser.rs

```diff
@@ -636,6 +636,77 @@ class Foo(A, B):
         insta::assert_debug_snapshot!(ast::Suite::parse(source, "<test>").unwrap());
     }
 
+    #[test]
+    #[cfg(feature = "all-nodes-with-ranges")]
+    fn test_parse_class_with_generic_type() {
+        let source = "\
+class Foo[T](A, B):
+  pass
+";
+        insta::assert_debug_snapshot!(ast::Suite::parse(source, "<test>").unwrap());
+    }
+
+    #[test]
+    #[cfg(feature = "all-nodes-with-ranges")]
+    fn test_parse_class_with_generic_type_with_bound() {
+        let source = "\
+class Foo[T: str](A, B):
+  pass
+";
+        insta::assert_debug_snapshot!(ast::Suite::parse(source, "<test>").unwrap());
+    }
+
+    #[test]
+    #[cfg(feature = "all-nodes-with-ranges")]
+    fn test_parse_class_with_generic_type_with_tuple_bound() {
+        let source = "\
+class Foo[T: (str, bytes)](A, B):
+  pass
+";
+        insta::assert_debug_snapshot!(ast::Suite::parse(source, "<test>").unwrap());
+    }
+
+    #[test]
+    #[cfg(feature = "all-nodes-with-ranges")]
+    fn test_parse_class_with_multiple_generic_types() {
+        let source = "\
+class Foo[T, U](A, B):
+  pass
+";
+        insta::assert_debug_snapshot!(ast::Suite::parse(source, "<test>").unwrap());
+    }
+
+    #[test]
+    #[cfg(feature = "all-nodes-with-ranges")]
+    fn test_parse_class_with_generic_type_var_tuple() {
+        let source = "\
+class Foo[*U](A, B):
+  pass
+";
+        insta::assert_debug_snapshot!(ast::Suite::parse(source, "<test>").unwrap());
+    }
+
+
+    #[test]
+    #[cfg(feature = "all-nodes-with-ranges")]
+    fn test_parse_class_with_generic_param_spec() {
+        let source = "\
+class Foo[**P](A, B):
+  pass
+";
+        insta::assert_debug_snapshot!(ast::Suite::parse(source, "<test>").unwrap());
+    }
+
+    #[test]
+    #[cfg(feature = "all-nodes-with-ranges")]
+    fn test_parse_class_with_all_possible_generic_types() {
+        let source = "\
```

Comment:
Can we add a test for trailing comma (assuming that's allowed)?

Reply:
Will determine and add test; I'd also like to check for a test corpus in CPython since I just made these simple cases up.

Reply:
https://github.com/RustPython/Parser/pull/95/commits/3ec64e124aa03c52136cfc5761d98cb021252943